src/Mini-Project-UI.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+import sys
+import pydicom
+from PySide6 import QtWidgets
+from PySide6.QtGui import QPixmap, QImage
+from PySide6.QtCore import QTimer
+from PIL import Image
+import os
+
+def dicom_image_opener(ds):
+    """Returns an image to be displayed by PySide6"""
+    picture = ds.pixel_array
+    picture = picture.astype(np.float32)
```

Comment:
re-using the same variable name makes it more difficult to understand the data type involved.
While some polymorphism is helpful, you are deliberately converting between data types here (vs. "taking what you get"), so it might be clearer if there were different variables involved.  If you are worried about memory, you can del(ete) the variable no longer in use to let the python GC free the memory.

Reply:
speaking of which, explicit data typing is helpful e.g.
def dicom_image_opener(ds: pydicom.Dataset) -> QImage:
...
etc.  You can type your variables as well, including having them be potentially a variety of types using the Union operator or | symbol.